src/App.js

```diff
@@ -1,43 +1,106 @@
 import React from "react";
 import "./App.css";
-import { makeShuffledDeck } from "./utils.js";
+import { makeShuffledDeck } from "./utils";
+import { imageMap } from "./images/images";
+
+const rank_names = {
+  2: "two",
+  3: "three",
+  4: "four",
+  5: "five",
+  6: "six",
+  7: "seven",
+  8: "eight",
+  9: "nine",
+  10: "ten",
+  11: "jack",
+  12: "queen",
+  13: "king",
+  14: "ace",
+};
 
 class App extends React.Component {
   constructor(props) {
-    // Always call super with props in constructor to initialise parent class
     super(props);
     this.state = {
-      // Set default value of card deck to new shuffled deck
       cardDeck: makeShuffledDeck(),
-      // currCards holds the cards from the current round
       currCards: [],
+      roundWinner: null,
+      playerOneWins: 0,
+      playerTwoWins: 0,
+      roundsRemaining: null,
     };
   }
 
   dealCards = () => {
+    //create the array for comparison
+    const newCards = this.state.cardDeck.slice(-2);
+
+    let currentRoundWinner = null;
```

Comment:
1. learnt something new about enumeration! this was really helpful
2. i originally tried to create a separate function (originally even in utils.js) to check for winner, but I had difficulty getting the function to work cos I was trying to access currCards (within this.state) via this.state.currCards, but looking at your suggested code as well as the reference solution, I see the work around is to create a new const array and to access this array within the function itself, instead of trying to access the currCards within the state.

however, not too sure why functions don't fully work by trying to accessing arrays within this.state